src/detection/Footprint.cc

```diff
@@ -219,11 +219,11 @@ class LegacyFootprintPersistenceHelper {
 
 std::pair<afw::table::Schema&, table::Key<int>&> spanSetPersistenceHelper() {
     static afw::table::Schema spanSetIdSchema;
-    static int initialize = true;
+    static int initialize = 1;
     static table::Key<int> idKey;
-    if (initialize) {
+    if (initialize != 0) {
         idKey = spanSetIdSchema.addField<int>("id", "id of the SpanSet catalog");
-        initialize = false;
+        initialize = 0;
```

Comment:
Why isn't `initialize` a boolean? Did `clang-tidy` detect a different usage somewhere?

Reply:
I don't know why it isn't a boolean. The check detected that it was an `int` and thus replaced the initialization. But we probably want to see why it isn't `bool` in the first place.